Octokit/Clients/Enterprise/EnterpriseAdminStatsClient.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Octokit
+{
+    /// <summary>
+    /// A client for GitHub's Enterprise Admin Stats API
+    /// </summary>
+    /// <remarks>
+    /// See the <a href="https://developer.github.com/v3/enterprise/admin_stats/">Enterprise Admin Stats API documentation</a> for more information.
+    ///</remarks>
+    public class EnterpriseAdminStatsClient : ApiClient, IEnterpriseAdminStatsClient
+    {
+        public EnterpriseAdminStatsClient(IApiConnection apiConnection)
+            : base(apiConnection)
+        { }
+
+        /// <summary>
+        /// Gets GitHub Enterprise statistics (must be Site Admin user).
+        /// </summary>
+        /// <remarks>
+        ///https://developer.github.com/v3/enterprise/admin_stats/#get-statistics
+        /// </remarks>
+        /// <returns>The <see cref="AdminStats"/> collection for the requested <see cref="AdminStatsType"/> type.</returns>
+        [SuppressMessage("Microsoft.Globalization", "CA1308:NormalizeStringsToUppercase")]
+        public Task<AdminStats> GetStatistics(AdminStatsType type)
+        {
+            var endpoint = ApiUrls.EnterpriseAdminStats(type.ToString().ToLowerInvariant());
+
+            if (type == AdminStatsType.All)
+            {
+                return ApiConnection.Get<AdminStats>(endpoint);
+            }
+
+            AdminStatsRepos repos = null;
+            AdminStatsHooks hooks = null;
+            AdminStatsPages pages = null;
+            AdminStatsOrgs orgs = null;
+            AdminStatsUsers users = null;
+            AdminStatsPulls pulls = null;
+            AdminStatsIssues issues = null;
+            AdminStatsMilestones milestones = null;
+            AdminStatsGists gists = null;
+            AdminStatsComments comments = null;
+
+            switch (type)
+            {
+                case AdminStatsType.Repos:
+                    {
+                        repos = ApiConnection.Get<AdminStatsRepos>(endpoint).Result;
```

Comment:
Could we swap these calls to `await ApiConnection.Get<AdminStatsRepos>(endpoint)`?

Might also need a `.ConfigureAwait(false)` if we're not doing this inside the connection...